apps/backoffice-v2/src/pages/Entity/components/Case/hooks/useDocumentsToolbar/useDocumentsToolbar.tsx

```diff
@@ -0,0 +1,63 @@
+import { useCallback, useLayoutEffect, useMemo, useRef } from 'react';
+
+import { useEntity } from '@/pages/Entity/hooks/useEntity/useEntity';
+import { BroadcastChannel } from 'broadcast-channel';
+
+interface IUseDocumentsToolbarProps {
+  imageId: string;
+  hideOpenExternalButton?: boolean;
+  onOpenDocumentInNewTab: (id: string) => void;
+}
+
+export const useDocumentsToolbar = ({
+  imageId,
+  hideOpenExternalButton,
+  onOpenDocumentInNewTab,
+}: IUseDocumentsToolbarProps) => {
+  const { workflow } = useEntity();
+
+  const broadcastChannel = useMemo(
+    () =>
+      new BroadcastChannel('openImageInNewTab', {
+        webWorkerSupport: false,
+      }),
+    [],
+  );
+
+  const isDocumentTabOpen = useRef(false);
+
+  const handler = useCallback(({ type }: { type: string }) => {
+    if (type === 'openImageInNewTabAck') {
```

Comment:
Might be premature but it seems like a good idea to have an enum for this event. Just to be used once for creating the channel and once for consuming it. (i.e. type === `BroadcastChannelEnum.OPEN_IMAGE_IN_NEW_TAB_ACK`)